worlds/pokemon_emerald/options.py

```diff
@@ -450,6 +611,48 @@ class BlindTrainers(Toggle):
     display_name = "Blind Trainers"
 
 
+class PurgeSpinners(Toggle):
+    """
+    Trainers will rotate in predictable patterns on a set interval instead of randomly and don't turn toward you when you run
+    """
+    display_name = "Purge Spinners"
+
+
+class MatchTrainerLevels(Choice):
+    """
+    When you start a battle with a trainer, your party's levels will be automatically set to match that trainer's highest level pokemon.
+
+    The experience you receive will match your party's average actual level, and will only be awarded when you win the battle.
+
+    This is a pseudo-replacement for a level cap and makes every trainer battle a fair fight while still allowing you to level up.
+
+    Off: The vanilla experience
+    Additive: The modifier you apply to your team is a flat bonus
+    Multiplicative: The modifier you apply to your team is a percent bonus
+    """
+    display_name = "Match Trainer Levels"
+    default = 0
+    option_off = 0
+    option_additive = 1
+    option_multiplicative = 2
+
+
+class MatchTrainerLevelsBonus(Range):
```

Comment:
This isn't a suggestion, but in case you weren't aware, negative numbers don't currently work for some of the range options in yamls. For example, you cannot do `random-range--25-25`. I doubt you'd want to add a new toggle option just to decide if the value is positive or negative, and, at worst, I would call it a limitation/downfall of Archipelago and not this world. Just figured you might want to know about this if you didn't know already

Reply:
Hmm. Good to keep in mind, but you're right, I'd rather not bend around limitations of the interface for comparatively niche use cases.